src/libsyntax/feature_gate.rs

```diff
@@ -1595,6 +1598,10 @@ impl<'a> Visitor<'a> for PostExpansionVisitor<'a> {
                 gate_feature_post!(&self, associated_type_defaults, ti.span,
                                   "associated type defaults are unstable");
             }
+            _ if ti.generics.is_parameterized() => {
```

Comment:
This is applying now to *all* kinds of trait items besides methods and types with a default value. This strikes me as a bit confusing. I think we should refactor this to be more like:

```
match ti.node {
    ast::TraitItemKind::Method(ref sig, ref block) => { .. /* as today */ .. }
    ast::TraitItemKind::Type(_, default) => {
        if default.is_some() {
                 gate_feature_post!(&self, associated_type_defaults, ti.span,
                                   "associated type defaults are unstable");
        }
        if ti.generics.is_parameterized() {
            gate_feature_post!(&self, generic_associated_types, ti.span,
                                   "generic associated types are unstable");
        }
    }
    _ => { /* nothing unstable here */ }
}
```